bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilder.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing.type;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.EventDescription;
+import org.eclipse.smarthome.core.types.StateDescription;
+
+/**
+ * Builder for {@link ChannelType}s
+ *
+ * @author Stefan Triller- Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ChannelTypeBuilder {
+
+    // mandatory
+    private @Nullable String bindingId;
```

Comment:
you could add the mandatory properties to the ChannelTypeBuilder constructor. This would make them NonNull. It also advices the consumer of the builder which properties are mandatory so the builder will only work if those are present.